### PR TITLE
Add initial version of kzg_7594 test generator

### PR DESCRIPTION
**Pull request by jtraglia**
This PR adds a new `kzg_7594` generator and an initial set of tests. The structure and a lot of the variables are copied from the `kzg_4844` generator. We can add more tests later, after this is merged.

Depends on https://github.com/ethereum/consensus-specs/pull/3684, it must be merged prior to this.

Edit: I can confirm that c-kzg passes these reference tests. Added in [this commit](https://github.com/ethereum/c-kzg-4844/commit/ab56ac63bb2cc41ce48eb08cf31d932ed401583b).

### CODE REVIEW

File: tests/generators/kzg_7594/main.py [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+from eth2spec.utils import bls
+from eth2spec.test.helpers.constants import EIP7594
+from eth2spec.test.helpers.typing import SpecForkName
+from eth2spec.gen_helpers.gen_base import gen_runner, gen_typing
+from eth2spec.eip7594 import spec
+
+
+###############################################################################
+# Helper functions
+###############################################################################
```
Comment (hwwhww) [inline tests/generators/kzg_7594/main.py:19]: probably good to move the 4844 duplicate helpers / constants to a `tests/core/pyspec/eth2spec/test/utils/kzg_tests.py` and import everything from it.

**Review by hwwhww — APPROVED** [review]
great work!

I mainly check if the folder names are correct. looks solid.

File: tests/generators/kzg_7594/main.py [hunk 2]
```diff
@@ -0,0 +1,837 @@
...
+    # Edge case: All cells are missing
+    cell_ids, partial_cells = [], []
+    expect_exception(spec.recover_all_cells, cell_ids, partial_cells)
+    identifier = make_id(cell_ids, partial_cells)
+    yield f'recover_all_cells_case_invalid_all_cells_are_missing_{identifier}', {
+        'input': {
+            'cell_ids': cell_ids,
+            'cells': encode_hex_list(partial_cells),
+        },
+        'output': encode_hex_list(recovered_cells)
```
Comment (asn-d6) [inline tests/generators/kzg_7594/main.py:695]: Shouldn't this be `'output': None` since it's an invalid test?

**Review by asn-d6 — APPROVED** [review]
Great work as always!

**Comment by jtraglia** [comment]
Thanks @asn-d6 😄

I can confirm that c-kzg-4844:das passes all of these tests.
